ocaml/xapi-client/tasks.ml

```diff
@@ -30,7 +30,7 @@ let wait_for_all_inner ~rpc ~session_id ~all_timeout ~tasks =
   let timeout_span =
     match all_timeout with
     | Some t ->
-        Some (t *. Mtime.s_to_ns |> Int64.of_float |> Mtime.Span.of_uint64_ns)
+        Some Mtime.Span.(Float.to_int (t *. 1e9) * ns)
```

Comment:
Is this more complicated than necessary? `t` is in seconds, so this could be `Span.(t * s)`

Reply:
t is a float, not an integer, so the easy conversion is not available. If, for some reason, the timeout is more finegrained than seconds we need to do the complicated conversion